Fabric.Authorization.AccessControl/src/app/navbar/navbar.component.ts

```diff
@@ -34,10 +34,55 @@ export class NavbarComponent implements OnInit {
     if (user && user.profile) {
       if (user.profile.family_name && user.profile.given_name) {
         return (user.profile.given_name + ' ' + user.profile.family_name);
-      }else {
+      } else if (user.profile.idp === 'AzureActiveDirectory') {
+        return this.parseAzureADProfileName(user.profile);
+      } else {
         return user.profile.name;
       }
     }
     return '';
   }
+
+  /*
+  Function: parseAzureADProfileName
+    This is complicated code.  Meant to parse the Profile object of a user from
+    Azure AD. It will look to see if it is an array and decided if it is able
+    to find a non-email name.  If it cant find that non-email name then it
+    will use the User Principal name,  i.e. email@domain
+  Parameters: profile, the profile of the user from OIDC
+  Returns: string, the name to use in the nav bar for Azure AD
+  */
+  parseAzureADProfileName(profile: any): string {
+    // 1) if the profile does not have an array,
+    // then return the name completely
+    if (!Array.isArray(profile.name)) {
```

Comment:
not for sure on TS, but f you try to access var.Prop (as array) and prop is not present, it should just skip it right?  and then nothing will be returned.

Reply:
I see, so I need to make sure name exists too.

Reply:
i have added an if above saying, if no name property exists, or if it is null, return ''no name detected''